sovrin_node/persistence/idr_cache.py

```diff
@@ -91,18 +94,28 @@ def close(self):
         self._keyValueStorage.close()
 
     def currentBatchCreated(self, stateRoot):
```

Comment:
Yes, Test1 can be done. To Test2 there has to be scenario created which will result in calling `revert` in an incorrect order of batches, simply patching some method to deliberately call `onBatchCommitted` with wrong args only proves that an assertion check works, its ok to do it, but not worth investing time in. I would rather create a scenario that result in reverting batches in incorrect order and see failures much before this code is reached. Test3 yes can be done